File: apps/monitor/models/monitor.py
"""Monitor 허브 핵심 모델 (D-MONITOR-REBUILD, P2).

Monitor{scope} = 개인화 모니터링 대상 (내가 등록한 대상 + 내 규칙 + 상태 기억).
Claim = Monitor에 부착되는 주장·마감 (구 thesis 개념의 재정의).
"""
import uuid

from django.conf import settings
from django.contrib.postgres.fields import ArrayField
from django.db import models


class Monitor(models.Model):
    """사용자가 등록한 모니터링 대상. scope에 따라 종목/섹터/테마/펀드/시장을 가리킨다."""

    class Scope(models.TextChoices):
        MARKET = "market", "Market"
        SECTOR = "sector", "Sector"
        THEME = "theme", "Theme"
        FUND = "fund", "Fund"
        STOCK = "stock", "Stock"

    class Status(models.TextChoices):
        SETTING_UP = "setting_up", "Setting Up"
        ACTIVE = "active", "Active"
        PAUSED = "paused", "Paused"
        ARCHIVED = "archived", "Archived"

    class State(models.TextChoices):
        """상태기(state machine) 판정값 — status(사용자 의도)와 별개의 자동 판정."""

        WARMING_UP = "warming_up", "데이터 수집 중"
        ACTIVE = "active", "활성 관제 중"
        STRENGTHENING = "strengthening", "강화 추세"
        WEAKENING = "weakening", "약화 추세"
        CRITICAL = "critical", "주의 필요"
        NEEDS_REVIEW = "needs_review", "점검 필요"
        EXPIRED = "expired", "기간 만료"
        PAUSED = "paused", "일시정지"

    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    user = models.ForeignKey(
        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="monitors"
    )
    scope = models.CharField(max_length=16, choices=Scope.choices)
    # 정규화된 대상 참조: stock=심볼(대문자), sector=섹터키, theme=바스켓 id, fund=ETF 심볼, market=지수키
    target_ref = models.CharField(max_length=64)
    name = models.CharField(max_length=200)
    status = models.CharField(
        max_length=16, choices=Status.choices, default=Status.SETTING_UP
    )
    # 상태기 자동 판정 (thesis_state_machine 이식)
    current_state = models.CharField(
        max_length=20, choices=State.choices, default=State.WARMING_UP
    )
    # 관제 종료 목표일 (없으면 무기한 — 90일+ 시 needs_review)
    target_date_end = models.DateField(null=True, blank=True)
    # 마감 제안 (MON-P3-ALERT): danger(critical) 연속 거래일 카운터 + 10일↑ 제안 플래그.
    # 제안만 — 마감은 사용자 수동 확정(결정 3-B). 배지 UI는 CLOSE 트랙 몫.
    danger_streak = models.IntegerField(default=0)
    close_suggested = models.BooleanField(default=False)
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        ordering = ["-updated_at"]
        constraints = [
            models.UniqueConstraint(
                fields=["user", "scope", "target_ref"],
                name="uniq_monitor_user_scope_target",
            )
        ]
        indexes = [models.Index(fields=["user", "status"])]

    def __str__(self):
        return f"{self.name} [{self.scope}:{self.target_ref}]"


class Claim(models.Model):
    """Monitor에 부착되는 주장 + 마감. 검증 결과(outcome)로 회고를 남긴다."""

    class Status(models.TextChoices):
        ACTIVE = "active", "Active"
        RESOLVED = "resolved", "Resolved"

    class Outcome(models.TextChoices):
        PENDING = "pending", "Pending"
        VALIDATED = "validated", "Validated"          # 적중
        PARTIAL = "partial", "Partial"                # 부분적중 (MON-CLOSE-UI ④)
        INVALIDATED = "invalidated", "Invalidated"    # 빗나감
        INCONCLUSIVE = "inconclusive", "Inconclusive" # 엣지(버튼 미노출)
        EXPIRED = "expired", "기한만료"                # 기한 경과·진입 미도달 (D-TIMING-DECISIONS-5 ④-B)

    class PriceZone(models.TextChoices):
        """가격 구간축 — Claim 가격 파라미터 대비 현재 종가의 위치 (D-TIMING-DECISIONS-5 ③-B).

        state_machine(신호축)과 별개의 가격축. 순수 파생값(price_zone.resolve_zone) — 저장은
        전이 감지용 직전값(last_price_zone)만. 라벨은 매수 타이밍 행동어.
        """

        EXITED = "exited", "이탈"           # close ≤ stop_price
        ENTRY = "entry", "진입 구간"         # stop < close ≤ entry
        APPROACH = "approach", "접근"        # entry < close ≤ entry×(1+버퍼)
        WAITING = "waiting", "관망"          # 버퍼 초과 ~ target 미만
        OVERHEATED = "overheated", "과열"    # close ≥ target

    class ProposedVerdict(models.TextChoices):
        """시스템 제안 판정 (마감 시점 종합점수 밴드 매핑). 최종=outcome, 델타=캘리브레이션."""

        VALIDATED = "validated", "Validated"
        PARTIAL = "partial", "Partial"
        INVALIDATED = "invalidated", "Invalidated"
        EXPIRED = "expired", "기한만료"  # 기한 경과·진입 미도달 (D-TIMING-DECISIONS-5 ④-B)

    class FactorTag(models.TextChoices):
        """회고 공통 요인 태그 (고정 enum — 자유문자열 금지)."""

        TIMING = "timing", "타이밍"
        EXT_SHOCK = "ext_shock", "외부 충격"
        INDICATOR_NOISE = "indicator_noise", "지표 노이즈"
        LUCK = "luck", "운"

    class ScenarioType(models.TextChoices):
        """시나리오 모드 (D-HOLD-DECISIONS 6). 저장 축 — 수익/손실/중립은 파생 상태(매입가 vs 종가).

        new_entry(미보유·매수 타이밍) / hold(보유 관리·매입가 앵커) / add_on(추가 매수·평단 백로그).
        """

        NEW_ENTRY = "new_entry", "신규 매수"
        HOLD = "hold", "보유 관리"
        ADD_ON = "add_on", "추가 매수"

    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    monitor = models.ForeignKey(
        Monitor, on_delete=models.CASCADE, related_name="claims"
    )
    assertion = models.TextField(help_text="주장")
    deadline = models.DateField(null=True, blank=True, help_text="마감")

    # 시나리오 모드 (D-HOLD-DECISIONS) — 기존 10행 default=new_entry 무해 편입.
    scenario_type = models.CharField(
        max_length=16, choices=ScenarioType.choices, default=ScenarioType.NEW_ENTRY,
        help_text="시나리오 모드(신규 매수/보유 관리/추가 매수)",
    )
    # 보유 모드 확정 사실 (D-HOLD-DECISIONS 1·5) — entry_price(제안·미래)와 분리. hold zone 앵커.
    purchase_price = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="매입가(확정 사실)"
    )
    purchase_date = models.DateField(
        null=True, blank=True, help_text="매입일(보유 기간 원천)"
    )

    # 매수 시나리오 가격 파라미터 (D-TIMING-DECISIONS-5 ②-A, additive).
    # 전부 null=구 가설(가격 없는 Claim, 그대로 유효). 정밀도=리포 관례 shared stocks OHLC.
    entry_price = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="진입가"
    )
    target_price = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="목표가(익절)"
    )
    stop_price = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="손절가"
    )
    # 적정가 밴드 (⑤-A) — 수동 입력 기본, 가치평가 통로의 미래 착지점(스키마는 별도 결정).
    fair_value_low = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="적정가 하단"
    )
    fair_value_high = models.DecimalField(
        max_digits=15, decimal_places=4, null=True, blank=True, help_text="적정가 상단"
    )
    # 가격 구간축 상태 (③-B) — 전이 감지용 직전 구간 + 진입 최초 도달 시각(EXPIRED 판정·통계 원천).
    last_price_zone = models.CharField(
        max_length=16, choices=PriceZone.choices, null=True, blank=True,
        help_text="직전 가격 구간(전이 감지용)",
    )
    entry_reached_at = models.DateTimeField(
        null=True, blank=True, help_text="진입 구간 최초 도달 시각(1회 기록)"
    )

    status = models.CharField(
        max_length=16, choices=Status.choices, default=Status.ACTIVE
    )
    outcome = models.CharField(
        max_length=16, choices=Outcome.choices, default=Outcome.PENDING
    )
    # 마감 회고 (MON-CLOSE-UI Phase 1) — 전부 마감 액션(close)에서만 설정.
    proposed_verdict = models.CharField(
        max_length=16, choices=ProposedVerdict.choices, null=True, blank=True,
        help_text="시스템 제안 판정 (마감 전 없음)",
    )
    resolved_by = models.ForeignKey(
        settings.AUTH_USER_MODEL, null=True, blank=True,
        on_delete=models.SET_NULL, related_name="resolved_claims",
        help_text="마감 주체",
    )
    factor_tags = ArrayField(
        models.CharField(max_length=20, choices=FactorTag.choices),
        default=list, blank=True, help_text="회고 요인 태그(고정 enum)",
    )
    retro_memo = models.TextField(blank=True, default="", help_text="회고 선택 한 줄")
    created_at = models.DateTimeField(auto_now_add=True)
    resolved_at = models.DateTimeField(null=True, blank=True)

    class Meta:
        ordering = ["-created_at"]

    def __str__(self):
        return f"Claim({self.assertion[:30]}) @ {self.monitor_id}"

File: frontend/types/monitor.ts
// Monitor 허브 타입 (MON-P3, api/v1/monitor/ 계약 미러)

export type MonitorScope = 'market' | 'sector' | 'theme' | 'fund' | 'stock'
export type MonitorStatus = 'setting_up' | 'active' | 'paused' | 'archived'
export type MonitorState =
  | 'warming_up'
  | 'active'
  | 'strengthening'
  | 'weakening'
  | 'critical'
  | 'needs_review'
  | 'expired'
  | 'paused'

export type IndicatorType =
  | 'market_data'
  | 'macro'
  | 'sentiment'
  | 'technical'
  | 'custom'
export type SupportDirection = 'positive' | 'negative'

export type ClaimStatus = 'active' | 'resolved'
export type ClaimOutcome =
  | 'pending'
  | 'validated'
  | 'partial'
  | 'invalidated'
  | 'inconclusive'
  | 'expired' // 기한만료 (D-TIMING-DECISIONS-5 ④-B)

// 마감 시 사용자가 고를 수 있는 최종 판정 (inconclusive 제외 — 엣지). expired = 시스템 제안·선택 가능.
export type ProposedVerdict = 'validated' | 'partial' | 'invalidated' | 'expired'
// VerdictBadge 등 표시용 — ProposedVerdict + inconclusive(중립 표시).
export type Verdict = ProposedVerdict | 'inconclusive'

// 가격 구간축 (BE Claim.PriceZone 미러 — TIMING-P1).
export type PriceZone = 'exited' | 'entry' | 'approach' | 'waiting' | 'overheated'

// 시나리오 모드 (BE Claim.ScenarioType 미러 — HOLD-P1).
export type ScenarioType = 'new_entry' | 'hold' | 'add_on'

// 사다리 밴드/틱 (BE zone_display 완결 — FE 하드코딩 제거, HOLD-P1).
export interface ZoneBand {
  key: string
  tone: PriceZone // ZONE_TONE 조회 키
  active: boolean
}
export interface ZoneTick {
  label: string
  value: number
}

// zone_display (serializer, BE 완결 표시 메타 — FE 재계산 금지). 가격 3필드 없으면 null.
// bands/ticks/rows/marker/pnl 전부 BE 단일 소스(HOLD-P1). ?는 하위호환(구 응답 폴백 허용).
export interface ZoneDisplay {
  zone: PriceZone | null
  label: string | null
  close: number | null
  boundaries: {
    stop: number
    entry: number
    approach_ceiling: number
    target: number
  }
  // ── HOLD-P1 additive ──
  mode?: ScenarioType
  mode_label?: string | null
  pnl_pct?: number | null // hold: (close/purchase−1)×100
  stop_distance_pct?: number | null // 손절여유(%): (close−stop)/close×100 (MON-DETAIL-P1 T1a). close 없으면 null
  marker_fraction?: number | null // 현재가 위치 0~1
  anchor_fraction?: number | null // hold 매입가 마커 0~1 (new_entry는 null)
  bands?: ZoneBand[] // 색 밴드(new_entry 5·hold 4)
  ticks?: ZoneTick[] // 미니 사다리 하단 라벨
  rows?: ZoneTick[] // 수직 사다리 우측 라벨(위→아래)
}

// 정합 재계산 블록 (TIMING-P2.5) — 힌트용, 자동 개서 아님.
export interface Coherence {
  symbol: string
  sigma: number | null
  note: string
  basis?: string
  coherent_horizon_days?: number
  coherent_deadline?: string
  coherent_target?: string
  rr?: number | null
  error?: string
}

// L계열 가격 제안 + 정합 프리필 (GET /monitor/scenario-suggest/?symbol=).
// mode=hold + purchase_price 제공 시 보유 프리필(매입가·진입가 미제안 + 본전 회복 N주).
export interface ScenarioSuggest {
  available: boolean
  symbol: string
  mode?: ScenarioType
  close?: number
  support_low?: number
  resistance_high?: number
  entry_suggest?: number
  atr?: number | null
  sigma?: number | null
  stop_suggest?: number | null
  target_suggest?: number | null
  horizon_days?: number | null
  deadline_suggest?: string | null
  rr_suggest?: number | null
  omit?: string | null
  // ── hold 전용 ──
  purchase_price?: number
  in_profit?: boolean
  breakeven_weeks?: number | null
  captions?: {
    entry?: string | null
    stop?: string | null
    target?: string | null
    deadline?: string | null
    breakeven?: string | null
  }
  basis?: string
  // entry + (target|deadline) 쿼리 제공 시에만 (재계산)
  coherence?: Coherence
}
// 회고 공통 요인 태그 (고정 enum — BE Claim.FactorTag 미러).
export type FactorTag = 'timing' | 'ext_shock' | 'indicator_noise' | 'luck'
// 지표별 마감 결과 (BE ClaimIndicatorResult.Result 미러).
export type IndicatorResultValue = 'hit' | 'partial' | 'miss' | 'na'

// BE 엔진(arrow_calculator·state_machine)이 산출한 파생 표시값 — FE는 렌더만.
export interface MonitorDisplay {
  degree: number // 0~180
  color: string // hex
  label: string // 화살표 라벨
  phase: 'full_moon' | 'waxing' | 'half_moon' | 'waning' | 'new_moon'
  phase_label: string
  phase_icon: string
}

export interface Monitor {
  id: string
  scope: MonitorScope
  target_ref: string
  name: string
  status: MonitorStatus
  current_state: MonitorState
  target_date_end: string | null
  resolved_label: string | null
  // 리스트 카드용 파생값 (list/detail에서 채워짐, create 응답은 null)
  latest_score: number | null
  display: MonitorDisplay | null
  indicator_count: number | null
  // MON-P2A T3: 점수 커버리지 {유효, 전체}(예 6/9). 스냅샷/annotation 없으면 null.
  indicator_coverage: { sufficient: number; total: number } | null
  next_deadline: string | null
  has_claim: boolean
  // 파이프라인(엔진) 소유 파생값 — 사용자 입력 불가 (MON-P3-ALERT)
  close_suggested: boolean
  danger_streak: number
  created_at: string
  updated_at: string
  // 통화 표시 계약 (PART B-FE 숫자 표시 규약 '가') — BE가 채워줌, 없으면 FE는 'USD' 폴백.
  currency_code?: string
}

export interface MonitorIndicator {
  id: string
  monitor: string
  name: string
  indicator_type: IndicatorType
  support_direction: SupportDirection
  weight: number
  source_key: string
  epsilon: number | null
  window: number | null
  decay: number | null
  is_active: boolean
  is_paused: boolean
  override_score: number | null
  created_at: string
  updated_at: string
}

export interface Claim {
  id: string
  monitor: string
  assertion: string
  deadline: string | null
  status: ClaimStatus
  outcome: ClaimOutcome
  // 마감 회고 (MON-CLOSE-UI Phase 1) — 전부 close 액션에서만 채워짐(마감 전 null/빈값).
  proposed_verdict: ProposedVerdict | null
  resolved_by: number | null // user id만 옴 — 마감자=소유자=현재 뷰어(owner-scoping)
  factor_tags: FactorTag[]
  retro_memo: string
  // 마감 동결 스냅샷 (MON-CLOSE-UI P1.5) — resolved면 동결값, PENDING이면 null.
  closure_snapshot: ClosureSnapshotData | null
  // 시나리오 모드 (HOLD-P1). 기존 데이터 default=new_entry.
  scenario_type: ScenarioType
  // 매수 시나리오 가격 (TIMING-P1, DRF Decimal→문자열). 무가격 구 가설이면 전부 null.
  entry_price: string | null
  target_price: string | null
  stop_price: string | null
  // 보유 확정 사실 (HOLD-P1) — hold 모드에서만 채워짐.
  purchase_price: string | null
  purchase_date: string | null
  fair_value_low: string | null
  fair_value_high: string | null
  // 가격 구간축 (파이프라인 소유·read-only) + BE 완결 표시 메타.
  last_price_zone: PriceZone | null
  entry_reached_at: string | null
  zone_display: ZoneDisplay | null
  created_at: string
  resolved_at: string | null
}

// 마감 시점 불변 동결값 (BE ClosureSnapshot 노출). 동결 점수·지표·달 위상 표시 소스.
export interface ClosureSnapshotData {
  overall_score: number
  frozen_at: string
  payload: Record<string, unknown>
}

// GET /monitor/claims/{id}/close-preview/ — 마감 모달 프리필(무상태, 읽기 전용).
export interface ClosePreviewIndicator {
  id: string
  name: string
  latest_value: number | null
  is_sufficient?: boolean // MON-P2A T3: 신호 패널 sufficient 배지 소스
}

export interface ClosePreview {
  proposed_verdict: ProposedVerdict
  overall_score: number
  indicators: ClosePreviewIndicator[]
}

// POST /monitor/claims/{id}/close/ 요청 바디.
export interface IndicatorResultInput {
  indicator_id: string
  result: IndicatorResultValue
}

export interface CloseClaimInput {
  final_verdict: ProposedVerdict
  factor_tags: FactorTag[]
  retro_memo?: string
  indicator_results: IndicatorResultInput[]
}

export interface WeakestLink {
  indicator_id: string
  indicator_name: string
  score: number
}

export interface EvaluateResult {
  monitor_id: string
  asof_date: string
  overall_score: number
  state: MonitorState
  state_changed: boolean
  reminder_needed: boolean
  data_coverage: number
  indicator_scores: Record<string, number>
  weakest_link: WeakestLink | null
  divergence: boolean
  bias_warning: { message: string } | null
  category_overlap: { message: string } | null
  snapshot_id: string
}

// Monitor 생성/수정 페이로드 (읽기 전용 필드 제외)
export interface MonitorInput {
  scope: MonitorScope
  target_ref: string
  name: string
  status?: MonitorStatus
  target_date_end?: string | null
}

// 지표 카탈로그 항목 (GET /monitor/catalog/). S계열 6종은 신규 메타 포함(TIMING-P1).
export type EvidenceStrength = 'strong' | 'medium' | 'weak'

export interface CatalogEntry {
  key: string
  name: string
  indicator_type: IndicatorType
  default_direction: SupportDirection
  source: string
  unit: string
  description: string
  // S계열 전용 메타 (기존 3종은 undefined).
  evidence_strength?: EvidenceStrength
  scoring_mode?: 'zscore' | 'bounded'
  default_selected?: boolean
  compute_key?: string
}

// ── 전이 알림 (MON-P3-ALERT, GET /monitor/alerts/) ──
// 억제(is_suppressed) 알림은 서버가 목록에서 이미 제외 → 개별 행에서 항상 false로 관측됨.
export interface AlertEvent {
  id: string
  monitor: string
  monitor_name: string
  target_ref: string
  from_state: MonitorState
  to_state: MonitorState
  from_label: string
  to_label: string
  asof: string
  score: number
  is_deterioration: boolean
  is_suppressed: boolean
  read: boolean
  created_at: string
}

// 헤더 벨 배지용 (GET /monitor/alerts/summary/) — 악화만 카운트.
export interface AlertSummary {
  unread_deterioration_count: number
}

// ── 상태밴드 스파크라인 (MON-P3-ALERT §6, GET /monitor/monitors/{id}/sparkline/) ──
export type SparklinePhase = 'full_moon' | 'waxing' | 'half_moon' | 'waning' | 'new_moon'

export interface SparklinePoint {
  asof: string
  score: number
}

// min/max는 score 정의역 [-1,1] 구간 — BE 엔진(score_to_phase)이 단일 출처, FE 하드코딩 금지.
export interface SparklineBand {
  phase: SparklinePhase
  label: string
  min: number
  max: number
}

export interface SparklineResponse {
  series: SparklinePoint[]
  bands: SparklineBand[]
  transitions: string[] // AlertEvent asof(ISO) 목록
  delta_5d: number | null // 계산값만 통과 — 표시는 회전 맵 트랙 몫(이번 트랙은 미표시)
  window: number
}

// ── 점수 정본 시계열 (MON-P2B T1, GET /monitor/monitors/{id}/snapshots/) ──
// MonitorSnapshot(동결 기록) 기반 — 스트립 델타·일지 스냅샷의 단일 원천.
// sparkline(추세 곡선, 재산출)과 별개 소스 — 혼동 금지.
export interface SnapshotSeriesPoint {
  asof: string
  score: number
  delta: number | null // 직전 스냅샷 대비 Δ (전체 첫 점만 null, ±0도 유효값)
}

export interface SnapshotSeriesResponse {
  series: SnapshotSeriesPoint[]
  window: number
}

// ── ADVISOR 브리핑 일지 (MON-P4-LA T3, GET /monitor/monitors/{id}/advisor_notes/) ──
// surface=L-A 한정, 최신순(asof desc, BE order_by('-asof')). 일지 kind=advisor 소스.
export interface AdvisorNote {
  id: string
  asof: string
  surface: string
  headline: string
  body: string
  coverage_n: number
  coverage_total: number
  model_id: string
  prompt_version: string
  created_at: string
}

// ── Claim 근거 (RECON-SWAP-0813 PART 1, /monitor/claim-evidences/) ──
// BE ClaimEvidence 미러 — 자동형(kind=auto, 지표 z 임계 대조)·수동형(kind=manual, 서술+재확인) 단일 테이블.
export type EvidenceKind = 'auto' | 'manual'
export type EvidenceOperator = 'gte' | 'lte' | 'gt' | 'lt'

export interface ClaimEvidence {
  id: string
  claim: string
  kind: EvidenceKind
  // 자동형 전용(수동형은 전부 null)
  indicator: string | null
  operator: EvidenceOperator | null
  threshold: number | null
  grace_days: number
  // 수동형 전용(자동형은 description=''·recheck_period_days=기본값)
  description: string
  recheck_period_days: number
  last_confirmed_at: string | null
  created_at: string
}

// POST /monitor/claim-evidences/ 요청 바디(모드별 필수값은 BE validate가 최종 판정).
export interface ClaimEvidenceInput {
  claim: string
  kind: EvidenceKind
  indicator?: string | null
  operator?: EvidenceOperator | null
  threshold?: number | null
  grace_days?: number
  description?: string
  recheck_period_days?: number
  last_confirmed_at?: string | null
}

// GET /monitor/claims/{id}/evidence-status/?as_of= — 근거 생사 판정(계약층, 읽기 전용·상태변경 없음).
export type EvidenceLifeStatus = 'alive' | 'dead' | 'expired' | 'unknown'

export interface EvidenceStatusResult {
  evidence_id: string
  kind: EvidenceKind
  status: EvidenceLifeStatus
  dead_streak_days: number
  indicator_name: string | null
  description: string
}

export interface EvidenceStatusResponse {
  claim_id: string
  as_of: string
  total: number
  alive: number
  results: EvidenceStatusResult[]
}

// ── 교체 검토 보류 이력 (RECON-SWAP-0813 PART 3-BE, /monitor/swap-hold-logs/) ──
// 횟수·누적일수는 이 로그를 조회 측(FE)이 집계한다 — 별도 카운터 없음(BE 단일 소스=로그).
export interface SwapHoldLog {
  id: string
  claim: string
  held_at: string
  candidate_ref: string | null
  // 보류 시점 가격 스냅샷 (P-1 성과 앵커, DRF Decimal→문자열). 구 기록(스냅샷 신설 이전)은 null.
  hold_price: string | null
  candidate_price: string | null
  note: string
  // PART C-4: BE 서버 산출 성과(held_at 스냅샷 → 최신 종가, DailyPrice 기반). 가격 이력·후보
  // 미지정이면 null — FE는 존재하면 우선 소비하고 없으면 Wallet 현재가 계산으로 폴백한다.
  hold_performance_pct?: number | null
  candidate_performance_pct?: number | null
}

export interface SwapHoldLogInput {
  claim: string
  candidate_ref?: string | null
  hold_price?: string | null
  candidate_price?: string | null
  note?: string
}

// ── 결정 일지 (RECON-SWAP-0813 PART 3-BE, /monitor/decision-journal-entries/) ──
// 마감/재커밋 실행 시 강제되는 1문장. sentence 품질 검증 없음(ADR §6 — 최소 글자수 등 FE도 금지).
export type DecisionJournalKind = 'close' | 'recommit' | 'hold'

export interface DecisionJournalEntry {
  id: string
  claim: string
  kind: DecisionJournalKind
  sentence: string
  created_at: string
}

export interface DecisionJournalEntryInput {
  claim: string
  kind: DecisionJournalKind
  sentence: string
}
